common/src/main/java/org/opensearch/ml/common/connector/McpConnector.java

```diff
@@ -0,0 +1,453 @@
+                    break;
+                case CREATED_TIME_FIELD:
+                    createdTime = Instant.ofEpochMilli(parser.longValue());
+                    break;
+                case LAST_UPDATED_TIME_FIELD:
+                    lastUpdateTime = Instant.ofEpochMilli(parser.longValue());
+                    break;
+                case CLIENT_CONFIG_FIELD:
+                    connectorClientConfig = ConnectorClientConfig.parse(parser);
+                    break;
+                case TENANT_ID_FIELD:
+                    tenantId = parser.textOrNull();
+                    break;
+                case URL_FIELD:
+                    url = parser.textOrNull();
+                    break;
+                case HEADERS_FIELD:
+                    headers = new HashMap<>();
+                    headers.putAll(parser.mapStrings());
+                    break;
+                default:
+                    parser.skipChildren();
+                    break;
+            }
+        }
+    }
+
+    protected Map<String, String> createDecryptedHeaders(Map<String, String> headers) {
+        if (headers == null) {
+            return null;
+        }
+        Map<String, String> decryptedHeaders = new HashMap<>();
+        StringSubstitutor substitutor = new StringSubstitutor(getDecryptedCredential(), "${credential.", "}");
+        for (Map.Entry<String, String> entry : headers.entrySet()) {
+            decryptedHeaders.put(entry.getKey(), substitutor.replace(entry.getValue()));
+        }
+        return decryptedHeaders;
+    }
+
+    @Override
+    public void decrypt(String action, BiFunction<String, String, String> function, String tenantId) {
+        Map<String, String> decrypted = new HashMap<>();
+        for (String key : credential.keySet()) {
+            decrypted.put(key, function.apply(credential.get(key), tenantId));
+        }
+        this.decryptedCredential = decrypted;
+        this.decryptedHeaders = createDecryptedHeaders(headers);
+    }
+
+    @Override
+    public void encrypt(BiFunction<String, String, String> function, String tenantId) {
+        for (String key : credential.keySet()) {
+            String encrypted = function.apply(credential.get(key), tenantId);
+            credential.put(key, encrypted);
+        }
+    }
+
+    @Override
+    public Connector cloneConnector() {
+        try (BytesStreamOutput bytesStreamOutput = new BytesStreamOutput()) {
+            this.writeTo(bytesStreamOutput);
+            StreamInput streamInput = bytesStreamOutput.bytes().streamInput();
+            return new McpConnector(streamInput);
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    public McpConnector(StreamInput input) throws IOException {
+        this.protocol = input.readString();
+        parseFromStream(input);
+    }
+
+    private void parseFromStream(StreamInput input) throws IOException {
+        Version streamInputVersion = input.getVersion();
```

Comment:
Where is this `streamInputVersion` being used ? Suggest remove this line if not being used